### DISCUSSION TITLE
Cannot build LSP4E locally after NonNullByDefault has been enabled

### DISCUSSION THREAD

**Discussion by ghentschke** [eclipse/lsp4e#1101] [Q&A] [answered]
After nonNullByDefault has been enabled with #1009 for the LSP4E project, I cannot build the project locally on my machine. I get a lot of Null pointer access errors, e.g.:
![image](https://github.com/user-attachments/assets/ec80051e-d4e1-46e8-a68b-939353a38bfc)
What has to be set in my IDE to build it locally again?

**Comment by ghentschke** [comment]
@sebthom can you give me a hint please?

**Reply by sebthom** [reply, marked as answer]
1. Ensure you are using the current platform target https://github.com/eclipse/lsp4e/blob/main/target-platforms/target-platform-latest/target-platform-latest.target
1. Ensure you have the EEA files jar on class path:
   ![image](https://github.com/user-attachments/assets/4222e659-175a-4692-9f8f-c22e04b9bac4)
3. Ensure `search for external annotations on all build paths` is enabled
   ![image](https://github.com/user-attachments/assets/20879800-d234-4638-8f09-77f6f05fd83e)

If that still is not enough check your Eclipse version. I am using 2024-06.

**Comment by ghentschke** [comment]
It works! Thank you for the fast response @sebthom !
